src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -171,6 +171,42 @@ def get_component_contexts(
                 output.append(context_item)
         return output
 
+    def get_installation_component_contexts(
+        self,
+        *,
+        filter: SentryAppInstallationFilterArgs,
+        component_type: str,
+        include_contexts_with_component: bool = True,
+    ) -> list[RpcSentryAppComponentContext]:
+        install_query = self._FQ.query_many(filter=filter)
+        app_ids = install_query.select_related("sentry_app", "sentry_app__application").values_list(
+            "sentry_app_id", flat=True
+        )
+        component_query = SentryAppComponent.objects.filter(
+            type=component_type, sentry_app_id__in=list(app_ids)
+        )
+        component_map: dict[int, SentryAppComponent] = {}
+        output = []
+
+        for component in component_query:
+            component_map[component.sentry_app.id] = component
```

Comment:
instead of `component.sentry_app.id` which will trigger another query to retrieve the related sentry app, you should be able to use `sentry_app_id` to just reference foreign key.